indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -128,6 +132,10 @@
 
   private volatile boolean started = false;
 
+  private final ScheduledExecutorService exec;
```

Comment:
dint get the benefit from making it listenable, do you propose to spilt the existing cleanup work into listeners ?